### ISSUE TITLE
Add Error when no sayings are tagged with actions

### ISSUE THREAD

**Issue by brtrndb** [samtecspg/articulate#610]
### Issues

Hello guys, I am just testing Articulate on my computer. I simply followed the the steps described in the docs (`docker-compose up`). I created an agent, then an intent with a few words. But when I click on the train button, the api throws an error:

```
api_1            | Debug: internal, implementation, error 
api_1            |     Error: handler method did not return a value, a promise, or throw an error
api_1            |     at module.exports.internals.Manager.execute (/usr/src/app/node_modules/glue/node_modules/hapi/lib/toolkit.js:52:29)
api_1            |     at process._tickCallback (internal/process/next_tick.js:68:7)
```

It seems there is already an issue on it (#572) that has been closed.

Have a good day :)

- Did you download a release or pull from Master? Which release?
I downloaded from master, then checkout the release v0.21.1, v0.21.0 and v0.20.0.

- What operating system are you running?
I am running Ubuntu 18.10.

- Are you running locally or in the cloud?
Locally with docker-compose.

- Which browser are you using?
Firefox 65.0.1 (64 bits)

**Comment by wrathagom** [comment]
@brtrndb we weren't ever able to replicate the other issue to get it fixed. Would you be able to provide us with an export of your training data, maybe training your agent will help us find the unhandled branch of code.

**Comment by brtrndb** [comment]
No problem, here is the full steps I've done.

- In a terminal.

```console
> git clone https://github.com/samtecspg/articulate.git
Clonage dans 'articulate'...
remote: Enumerating objects: 54, done.
remote: Counting objects: 100% (54/54), done.
remote: Compressing objects: 100% (54/54), done.
remote: Total 24170 (delta 2), reused 30 (delta 0), pack-reused 24116
Réception d'objets: 100% (24170/24170), 31.15 MiB | 9.86 MiB/s, fait.
Résolution des deltas: 100% (12702/12702), fait.
> cd articulate/
> git checkout v0.21.1
Note : extraction de 'v0.21.1'.

Vous êtes dans l'état « HEAD détachée ». Vous pouvez visiter, faire des modifications
expérimentales et les valider. Il vous suffit de faire une autre extraction pour
abandonner les commits que vous faites dans cet état sans impacter les autres branches

Si vous voulez créer une nouvelle branche pour conserver les commits que vous créez,
il vous suffit d'utiliser « checkout -b » (maintenant ou plus tard) comme ceci :

  git checkout -b <nom-de-la-nouvelle-branche>

HEAD est maintenant sur f466614c Updates for 0.21.1 release
> docker-compose up
Creating network "articulate_default" with the default driver
Pulling api (samtecspg/articulate-api:0.21.1)...
0.21.1: Pulling from samtecspg/articulate-api
1c7fe136a31e: Pull complete
ece825d3308b: Pull complete
06854774e2f3: Pull complete
f0db43b9b8da: Pull complete
aa50047aad93: Pull complete
42b3631d8d2e: Pull complete
1571a77bec6f: Pull complete
437249b4f02b: Pull complete
f7e70415e37e: Pull complete
2934799bb384: Pull complete
a18ad7440cde: Pull complete
8e8792891647: Pull complete
0f8e35f07277: Pull complete
0ecd8bcd0a5f: Pull complete
12ba75844233: Pull complete
Pulling ui (samtecspg/articulate-ui:0.21.1)...
0.21.1: Pulling from samtecspg/articulate-ui
1c7fe136a31e: Already exists
ece825d3308b: Already exists
06854774e2f3: Already exists
f0db43b9b8da: Already exists
aa50047aad93: Already exists
42b3631d8d2e: Already exists
1571a77bec6f: Already exists
437249b4f02b: Already exists
fe18482c6786: Pull complete
d383a7cc151a: Pull complete
48c4ca291686: Pull complete
ba88fef5ae6a: Pull complete
9bf450a3ec93: Pull complete
99a7138e92b2: Pull complete
189d22ad8e62: Pull complete
Pulling rasa (samtecspg/articulate-rasa:0.21.1)...
0.21.1: Pulling from samtecspg/articulate-rasa
6ae821421a7d: Pull complete
759f8891d8a3: Pull complete
6e316fa27cc6: Pull complete
d8f53da79476: Pull complete
b69aaefd884f: Pull complete
4c51bd86ca03: Pull complete
a75706166c99: Pull complete
b934f2e45e93: Pull complete
760b5a14056d: Pull complete
cb8c0ecef471: Pull complete
dae7ee45a16b: Pull complete
634b3d16e08a: Pull complete
ee8971f36c8f: Pull complete
8f6b09a81542: Pull complete
e9a3c1e41135: Pull complete
d3f0a5d54632: Pull complete
3e5d91c92b29: Pull complete
7e4eb5e14f0d: Pull complete
Pulling duckling (samtecspg/duckling:0.1.6.0)...
0.1.6.0: Pulling from samtecspg/duckling
3d77ce4481b1: Pull complete
328eb84d4cb2: Pull complete
dd2377981932: Pull complete
7eae90b23610: Pull complete
8aa2219bfaa2: Pull complete
3316209e3290: Pull complete
434d03cbe6e1: Pull complete
3184a4686b82: Pull complete
Pulling redis (redis:4.0.6-alpine)...
4.0.6-alpine: Pulling from library/redis
ff3a5c916c92: Pull complete
aae70a2e6027: Pull complete
87c655da471c: Pull complete
2aa929812fd0: Pull complete
ed33a57da65e: Pull complete
dffac9e716e1: Pull complete
Pulling elasticsearch (docker.elastic.co/elasticsearch/elasticsearch:6.5.1)...
6.5.1: Pulling from elasticsearch/elasticsearch
aeb7866da422: Pull complete
8ed52286043f: Pull complete
8b87f38e78e9: Pull complete
a4cf659acf47: Pull complete
0385f485f7da: Pull complete
e76d98016114: Pull complete
00185d9e8607: Pull complete
14f38d200567: Pull complete
Creating articulate_redis_1         ... done
Creating articulate_ui_1            ... done
Creating articulate_rasa_1          ... done
Creating articulate_duckling_1      ... done
Creating articulate_elasticsearch_1 ... done
Creating articulate_api_1           ... done
Attaching to articulate_redis_1, articulate_ui_1, articulate_api_1, articulate_elasticsearch_1, articulate_rasa_1, articulate_duckling_1
redis_1          | 1:C 01 Mar 13:45:31.839 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
redis_1          | 1:C 01 Mar 13:45:31.839 # Redis version=4.0.6, bits=64, commit=00000000, modified=0, pid=1, just started
redis_1          | 1:C 01 Mar 13:45:31.839 # Configuration loaded
ui_1             | yarn run v1.13.0
redis_1          | 1:M 01 Mar 13:45:31.841 * Running mode=standalone, port=6379.
redis_1          | 1:M 01 Mar 13:45:31.841 # WARNING: The TCP backlog setting of 511 cannot be enforced because /proc/sys/net/core/somaxconn is set to the lower value of 128.
redis_1          | 1:M 01 Mar 13:45:31.841 # Server initialized
redis_1          | 1:M 01 Mar 13:45:31.841 # WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
redis_1          | 1:M 01 Mar 13:45:31.841 # WARNING you have Transparent Huge Pages (THP) support enabled in your kernel. This will create latency and memory usage issues with Redis. To fix this issue run the command 'echo never > /sys/kernel/mm/transparent_hugepage/enabled' as root, and add it to your /etc/rc.local in order to retain the setting after a reboot. Redis must be restarted after THP is disabled.
redis_1          | 1:M 01 Mar 13:45:31.841 * Ready to accept connections
api_1            | 2019/03/01 13:45:32 Waiting for: http://elasticsearch:9200
elasticsearch_1  | OpenJDK 64-Bit Server VM warning: Option UseConcMarkSweepGC was deprecated in version 9.0 and will likely be removed in a future release.
api_1            | 2019/03/01 13:45:32 Problem with request: Get http://elasticsearch:9200: dial tcp 172.24.0.5:9200: connect: connection refused. Sleeping 1s
ui_1             | $ cross-env NODE_ENV=production env-cmd .env node server
api_1            | 2019/03/01 13:45:33 Problem with request: Get http://elasticsearch:9200: dial tcp 172.24.0.5:9200: connect: connection refused. Sleeping 1s
ui_1             | Server started ! ✓
ui_1             | 
ui_1             | Access URLs:
ui_1             | -----------------------------------
ui_1             | Localhost: http://localhost:3000
ui_1             |       LAN: http://172.24.0.3:3000
ui_1             | -----------------------------------
ui_1             | Press CTRL-C to stop
ui_1             |     
api_1            | 2019/03/01 13:45:34 Problem with request: Get http://elasticsearch:9200: dial tcp 172.24.0.5:9200: connect: connection refused. Sleeping 1s
elasticsearch_1  | [2019-03-01T13:45:34,847][WARN ][o.e.c.l.LogConfigurator  ] [unknown] Some logging configurations have %marker but don't have %node_name. We will automatically add %node_name to the pattern to ease the migration for users who customize log4j2.properties but will stop this behavior in 7.0. You should manually replace `%node_name` with `[%node_name]%marker ` in these locations:
elasticsearch_1  |   /usr/share/elasticsearch/config/log4j2.properties
duckling_1       | no port specified, defaulting to port 8000
duckling_1       | Listening on http://0.0.0.0:8000
api_1            | 2019/03/01 13:45:35 Problem with request: Get http://elasticsearch:9200: dial tcp 172.24.0.5:9200: connect: connection refused. Sleeping 1s
elasticsearch_1  | [2019-03-01T13:45:35,732][INFO ][o.e.e.NodeEnvironment    ] [07b9WJy] using [1] data paths, mounts [[/usr/share/elasticsearch/data (/dev/sda1)]], net usable_space [160.1gb], net total_space [224.3gb], types [ext4]
elasticsearch_1  | [2019-03-01T13:45:35,733][INFO ][o.e.e.NodeEnvironment    ] [07b9WJy] heap size [494.9mb], compressed ordinary object pointers [true]
elasticsearch_1  | [2019-03-01T13:45:35,734][INFO ][o.e.n.Node               ] [07b9WJy] node name derived from node ID [07b9WJyJR5qJUbjj_uJm2g]; set [node.name] to override
elasticsearch_1  | [2019-03-01T13:45:35,735][INFO ][o.e.n.Node               ] [07b9WJy] version[6.5.1], pid[1], build[default/tar/8c58350/2018-11-16T02:22:42.182257Z], OS[Linux/4.18.0-15-generic/amd64], JVM[Oracle Corporation/OpenJDK 64-Bit Server VM/11.0.1/11.0.1+13]
elasticsearch_1  | [2019-03-01T13:45:35,735][INFO ][o.e.n.Node               ] [07b9WJy] JVM arguments [-Xms1g, -Xmx1g, -XX:+UseConcMarkSweepGC, -XX:CMSInitiatingOccupancyFraction=75, -XX:+UseCMSInitiatingOccupancyOnly, -XX:+AlwaysPreTouch, -Xss1m, -Djava.awt.headless=true, -Dfile.encoding=UTF-8, -Djna.nosys=true, -XX:-OmitStackTraceInFastThrow, -Dio.netty.noUnsafe=true, -Dio.netty.noKeySetOptimization=true, -Dio.netty.recycler.maxCapacityPerThread=0, -Dlog4j.shutdownHookEnabled=false, -Dlog4j2.disable.jmx=true, -Djava.io.tmpdir=/tmp/elasticsearch.R4ZfWAin, -XX:+HeapDumpOnOutOfMemoryError, -XX:HeapDumpPath=data, -XX:ErrorFile=logs/hs_err_pid%p.log, -Xlog:gc*,gc+age=trace,safepoint:file=logs/gc.log:utctime,pid,tags:filecount=32,filesize=64m, -Djava.locale.providers=COMPAT, -XX:UseAVX=2, -Des.cgroups.hierarchy.override=/, -Xms512m, -Xmx512m, -Des.path.home=/usr/share/elasticsearch, -Des.path.conf=/usr/share/elasticsearch/config, -Des.distribution.flavor=default, -Des.distribution.type=tar]
api_1            | 2019/03/01 13:45:36 Problem with request: Get http://elasticsearch:9200: dial tcp 172.24.0.5:9200: connect: connection refused. Sleeping 1s
rasa_1           | 2019-03-01 13:45:37+0000 [-] Log opened.
rasa_1           | 2019-03-01 13:45:37+0000 [-] Site starting on 5000
rasa_1           | 2019-03-01 13:45:37+0000 [-] Starting factory <twisted.web.server.Site object at 0x7fb9332feb00>
api_1            | 2019/03/01 13:45:37 Problem with request: Get http://elasticsearch:9200: dial tcp 172.24.0.5:9200: connect: connection refused. Sleeping 1s
elasticsearch_1  | [2019-03-01T13:45:37,527][INFO ][o.e.p.PluginsService     ] [07b9WJy] loaded module [aggs-matrix-stats]
elasticsearch_1  | [2019-03-01T13:45:37,528][INFO ][o.e.p.PluginsService     ] [07b9WJy] loaded module [analysis-common]
elasticsearch_1  | [2019-03-01T13:45:37,528][INFO ][o.e.p.PluginsService     ] [07b9WJy] loaded module [ingest-common]
elasticsearch_1  | [2019-03-01T13:45:37,528][INFO ][o.e.p.PluginsService     ] [07b9WJy] loaded module [lang-expression]
elasticsearch_1  | [2019-03-01T13:45:37,528][INFO ][o.e.p.PluginsService     ] [07b9WJy] loaded module [lang-mustache]
elasticsearch_1  | [2019-03-01T13:45:37,528][INFO ][o.e.p.PluginsService     ] [07b9WJy] loaded module [lang-painless]
elasticsearch_1  | [2019-03-01T13:45:37,528][INFO ][o.e.p.PluginsService     ] [07b9WJy] loaded module [mapper-extras]
elasticsearch_1  | [2019-03-01T13:45:37,528][INFO ][o.e.p.PluginsService     ] [07b9WJy] loaded module [parent-join]
elasticsearch_1  | [2019-03-01T13:45:37,528][INFO ][o.e.p.PluginsService     ] [07b9WJy] loaded module [percolator]
elasticsearch_1  | [2019-03-01T13:45:37,528][INFO ][o.e.p.PluginsService     ] [07b9WJy] loaded module [rank-eval]
elasticsearch_1  | [2019-03-01T13:45:37,529][INFO ][o.e.p.PluginsService     ] [07b9WJy] loaded module [reindex]
elasticsearch_1  | [2019-03-01T13:45:37,529][INFO ][o.e.p.PluginsService     ] [07b9WJy] loaded module [repository-url]
elasticsearch_1  | [2019-03-01T13:45:37,529][INFO ][o.e.p.PluginsService     ] [07b9WJy] loaded module [transport-netty4]
elasticsearch_1  | [2019-03-01T13:45:37,529][INFO ][o.e.p.PluginsService     ] [07b9WJy] loaded module [tribe]
elasticsearch_1  | [2019-03-01T13:45:37,529][INFO ][o.e.p.PluginsService     ] [07b9WJy] loaded module [x-pack-ccr]
elasticsearch_1  | [2019-03-01T13:45:37,529][INFO ][o.e.p.PluginsService     ] [07b9WJy] loaded module [x-pack-core]
elasticsearch_1  | [2019-03-01T13:45:37,529][INFO ][o.e.p.PluginsService     ] [07b9WJy] loaded module [x-pack-deprecation]
elasticsearch_1  | [2019-03-01T13:45:37,530][INFO ][o.e.p.PluginsService     ] [07b9WJy] loaded module [x-pack-graph]
elasticsearch_1  | [2019-03-01T13:45:37,530][INFO ][o.e.p.PluginsService     ] [07b9WJy] loaded module [x-pack-logstash]
elasticsearch_1  | [2019-03-01T13:45:37,530][INFO ][o.e.p.PluginsService     ] [07b9WJy] loaded module [x-pack-ml]
elasticsearch_1  | [2019-03-01T13:45:37,530][INFO ][o.e.p.PluginsService     ] [07b9WJy] loaded module [x-pack-monitoring]
elasticsearch_1  | [2019-03-01T13:45:37,530][INFO ][o.e.p.PluginsService     ] [07b9WJy] loaded module [x-pack-rollup]
elasticsearch_1  | [2019-03-01T13:45:37,530][INFO ][o.e.p.PluginsService     ] [07b9WJy] loaded module [x-pack-security]
elasticsearch_1  | [2019-03-01T13:45:37,530][INFO ][o.e.p.PluginsService     ] [07b9WJy] loaded module [x-pack-sql]
elasticsearch_1  | [2019-03-01T13:45:37,530][INFO ][o.e.p.PluginsService     ] [07b9WJy] loaded module [x-pack-upgrade]
elasticsearch_1  | [2019-03-01T13:45:37,531][INFO ][o.e.p.PluginsService     ] [07b9WJy] loaded module [x-pack-watcher]
elasticsearch_1  | [2019-03-01T13:45:37,531][INFO ][o.e.p.PluginsService     ] [07b9WJy] loaded plugin [ingest-geoip]
elasticsearch_1  | [2019-03-01T13:45:37,531][INFO ][o.e.p.PluginsService     ] [07b9WJy] loaded plugin [ingest-user-agent]
api_1            | 2019/03/01 13:45:38 Problem with request: Get http://elasticsearch:9200: dial tcp 172.24.0.5:9200: connect: connection refused. Sleeping 1s
api_1            | 2019/03/01 13:45:39 Problem with request: Get http://elasticsearch:9200: dial tcp 172.24.0.5:9200: connect: connection refused. Sleeping 1s
api_1            | 2019/03/01 13:45:40 Problem with request: Get http://elasticsearch:9200: dial tcp 172.24.0.5:9200: connect: connection refused. Sleeping 1s
elasticsearch_1  | [2019-03-01T13:45:40,826][INFO ][o.e.d.DiscoveryModule    ] [07b9WJy] using discovery type [single-node] and host providers [settings]
api_1            | 2019/03/01 13:45:41 Problem with request: Get http://elasticsearch:9200: dial tcp 172.24.0.5:9200: connect: connection refused. Sleeping 1s
elasticsearch_1  | [2019-03-01T13:45:41,409][INFO ][o.e.n.Node               ] [07b9WJy] initialized
elasticsearch_1  | [2019-03-01T13:45:41,410][INFO ][o.e.n.Node               ] [07b9WJy] starting ...
elasticsearch_1  | [2019-03-01T13:45:41,587][INFO ][o.e.t.TransportService   ] [07b9WJy] publish_address {172.24.0.5:9300}, bound_addresses {0.0.0.0:9300}
elasticsearch_1  | [2019-03-01T13:45:41,606][WARN ][o.e.b.BootstrapChecks    ] [07b9WJy] max virtual memory areas vm.max_map_count [65530] is too low, increase to at least [262144]
elasticsearch_1  | [2019-03-01T13:45:41,647][INFO ][o.e.h.n.Netty4HttpServerTransport] [07b9WJy] publish_address {172.24.0.5:9200}, bound_addresses {0.0.0.0:9200}
elasticsearch_1  | [2019-03-01T13:45:41,647][INFO ][o.e.n.Node               ] [07b9WJy] started
elasticsearch_1  | [2019-03-01T13:45:41,722][INFO ][o.e.g.GatewayService     ] [07b9WJy] recovered [0] indices into cluster_state
elasticsearch_1  | [2019-03-01T13:45:41,832][INFO ][o.e.l.LicenseService     ] [07b9WJy] license [a0b89731-21dc-40e2-9919-89cd5af7115a] mode [basic] - valid
api_1            | 2019/03/01 13:45:42 Received 200 from http://elasticsearch:9200
api_1            | yarn run v1.13.0
api_1            | $ nodemon server --exec babel-node
api_1            | [nodemon] 1.18.4
api_1            | [nodemon] reading config ./package.json
api_1            | [nodemon] to restart at any time, enter `rs`
api_1            | [nodemon] or send SIGHUP to 29 to restart
api_1            | [nodemon] ignoring: ./node_modules/**/*
api_1            | [nodemon] watching: *.*
api_1            | [nodemon] watching extensions: js,mjs,json
api_1            | [nodemon] starting `babel-node server`
api_1            | [nodemon] spawning
api_1            | [nodemon] child pid: 41
api_1            | [nodemon] watching 286 files
api_1            | Browserslist: caniuse-lite is outdated. Please run next command `yarn upgrade caniuse-lite browserslist`
elasticsearch_1  | [2019-03-01T13:45:48,104][WARN ][o.e.d.c.m.MetaDataCreateIndexService] [07b9WJy] the default number of shards will change from [5] to [1] in 7.0.0; if you wish to continue using the default of [5] shards, you must manage this on the create index request or with an index template
elasticsearch_1  | [2019-03-01T13:45:48,180][INFO ][o.e.c.m.MetaDataCreateIndexService] [07b9WJy] [document] creating index, cause [api], templates [], shards [5]/[1], mappings []
elasticsearch_1  | [2019-03-01T13:45:48,651][INFO ][o.e.c.m.MetaDataMappingService] [07b9WJy] [document/OOBTazgBSTqfTBG0avYMtw] create_mapping [document]
api_1            | Debug: internal, implementation, error 
api_1            |     Error: handler method did not return a value, a promise, or throw an error
api_1            |     at module.exports.internals.Manager.execute (/usr/src/app/node_modules/hapi/lib/toolkit.js:52:29)
api_1            |     at process._tickCallback (internal/process/next_tick.js:68:7)
```

- Then I opened Firefox on http://localhost:3000/ and created an agent.

![screenshot_2019-03-01 articulate1](https://user-images.githubusercontent.com/2642899/53643141-9c9c5300-3c33-11e9-95f7-1be6672436e1.png)

- I created an intent, and clicked on 'Train'.

![screenshot_2019-03-01 articulate2](https://user-images.githubusercontent.com/2642899/53643335-146a7d80-3c34-11e9-8bb8-56c17195e543.png)

And I get the error message in the chat panel with stacktrace message in the terminal.

Here the the json file exported for my agent.

```json
{
  "agentName": "MyAgent",
  "description": "Does stuffs.",
  "language": "en",
  "timezone": "UTC",
  "useWebhook": false,
  "usePostFormat": false,
  "multiCategory": false,
  "categoryClassifierThreshold": 0.5,
  "fallbackAction": "Default Fallback Action",
  "status": "Out of Date",
  "lastTraining": "",
  "extraTrainingData": false,
  "enableModelsPerCategory": false,
  "model": "",
  "parameters": {},
  "settings": {
    "rasaURL": "http://rasa:5000",
    "ducklingURL": "http://duckling:8000",
    "ducklingDimension": [
      "amount-of-money",
      "distance",
      "duration",
      "email",
      "number",
      "ordinal",
      "phone-number",
      "quantity",
      "temperature",
      "time",
      "url",
      "volume"
    ],
    "spacyPretrainedEntities": [
      "PERSON",
      "NORP",
      "FAC",
      "ORG",
      "GPE",
      "LOC",
      "PRODUCT",
      "EVENT",
      "WORK_OF_ART",
      "LAW",
      "LANGUAGE",
      "DATE",
      "TIME",
      "PERCENT",
      "MONEY",
      "QUANTITY",
      "ORDINAL",
      "CARDINAL"
    ],
    "categoryClassifierPipeline": [
      {
        "name": "intent_featurizer_count_vectors"
      },
      {
        "name": "intent_classifier_tensorflow_embedding",
        "intent_tokenization_flag": true,
        "intent_split_symbol": "+__+"
      },
      {
        "name": "nlp_spacy"
      },
      {
        "name": "tokenizer_spacy"
      },
      {
        "name": "ner_crf"
      },
      {
        "name": "ner_synonyms"
      },
      {
        "name": "ner_spacy"
      }
    ],
    "sayingClassifierPipeline": [
      {
        "name": "intent_featurizer_count_vectors"
      },
      {
        "name": "intent_classifier_tensorflow_embedding",
        "intent_tokenization_flag": true,
        "intent_split_symbol": "+__+"
      },
      {
        "name": "nlp_spacy"
      },
      {
        "name": "tokenizer_spacy"
      },
      {
        "name": "ner_crf"
      },
      {
        "name": "ner_synonyms"
      },
      {
        "name": "ner_spacy"
      }
    ],
    "keywordClassifierPipeline": [
      {
        "name": "nlp_spacy"
      },
      {
        "name": "tokenizer_spacy"
      },
      {
        "name": "ner_crf"
      },
      {
        "name": "ner_synonyms"
      },
      {
        "name": "ner_spacy"
      }
    ],
    "sayingsPageSize": 5,
    "keywordsPageSize": 5,
    "modifierSayingsPageSize": 5,
    "reviewPageSize": 5
  },
  "categoryRecognizer": false,
  "modifiersRecognizer": false,
  "modifiersRecognizerJustER": "",
  "creationDate": "1551448109000",
  "modificationDate": "1551448152000",
  "actions": [
    {
      "actionName": "Default Fallback Action",
      "slots": [],
      "responses": [
        {
          "textResponse": "Sorry can you rephrase that?",
          "actions": []
        },
        {
          "textResponse": "I'm still learning to speak with humans. What you mean?",
          "actions": []
        }
      ],
      "useWebhook": false,
      "usePostFormat": false,
      "creationDate": "1551448109000",
      "modificationDate": "1551448109000"
    }
  ],
  "categories": [
    {
      "categoryName": "greetings",
      "enabled": true,
      "actionThreshold": 0.5,
      "status": "Out of Date",
      "lastTraining": "2019-03-01T13:48:45Z",
      "model": "",
      "extraTrainingData": true,
      "parameters": {},
      "creationDate": "1551448125000",
      "modificationDate": "1551448152000",
      "sayings": [
        {
          "userSays": "hi",
          "keywords": [],
          "actions": [],
          "creationDate": "1551448133000",
          "modificationDate": "1551448133000"
        },
        {
          "userSays": "hello",
          "keywords": [],
          "actions": [],
          "creationDate": "1551448137000",
          "modificationDate": "1551448137000"
        },
        {
          "userSays": "good morning",
          "keywords": [],
          "actions": [],
          "creationDate": "1551448152000",
          "modificationDate": "1551448152000"
        }
      ]
    }
  ],
  "keywords": []
}
```

I hope this will help you. If you need any more information, don't hesitate :)

**Comment by wrathagom** [comment]
@brtrndb one thing I notice is that you're not using categories and actions the way they need to be used. I am curious if that is actually the problem and we just need to improve the error message.

Can you watch this video real quick and see if it helps: https://www.youtube.com/watch?v=opL04qS6S5U

**Comment by brtrndb** [comment]
Oh ! The video explains everything, thanks a lot ! I did not create action for the user sayings. And now, everything seems to work very well.

But you're right, better error message in the chat panel would be nice. Or perhaps, a check before training if there is at least one action for each user entries.

Again, thanks a lot for your help :)

**Comment by wrathagom** [comment]
Closed by #611
